Lib/pathlib.py

```diff
@@ -825,7 +825,7 @@ def absolute(self):
         """
         if self.is_absolute():
             return self
-        return self._from_parts([self.cwd()] + self._parts)
+        return self._from_parts([os.getcwd()] + self._parts)
```

Comment:
This overlooks relative drive paths, such as "Z:spam". If `self._drv[1:2] == ':', the drive working directory can be resolved with `nt._getfullpathname(self._drv)`. Otherwise use `os.getcwd()`.

Reply:
If that's true, it's a bug in `ntpath.join()`, which we use to join arguments: https://github.com/python/cpython/blob/08e5594cf3d42391a48e0311f6b9393ec2e00e1e/Lib/pathlib.py#L293

Reply:
Nevermind, I see what you mean. Could you log it as a separate bug? It's not a regression in this patch.

Reply:
Okay. But the suggested fix should work in 3.12 now, right? This was discussed before on another issue, and it was blocked by a bug in `_from_parts()`. Fixing the bug in 3.10 and 3.11 will be more work.

Reply:
Gotcha, I hadn't considered that. Thank you. If you log a bug I'll mark this PR as a draft and get that change made first.